packages/next/src/state.tsx

```diff
@@ -0,0 +1,136 @@
+"use client";
+
+import type {
+  SfCart,
+  SfCurrency,
+  SfCustomer,
+} from "@vue-storefront/unified-data-model";
+import React, {
+  type PropsWithChildren,
+  createContext,
+  useContext,
+  useRef,
+} from "react";
+import { type StoreApi, createStore, useStore } from "zustand";
+
+export interface SfStateProps {
+  currencies: SfCurrency[];
+  currency: SfCurrency;
+  locale: string;
+  locales: string[];
+}
+
+interface SfState extends SfStateProps {
+  cart: SfCart | null;
+  customer: SfCustomer | null;
+  setCart: (cart: SfCart) => void;
+  setCurrencies: (currencies: SfCurrency[]) => void;
+  setCurrency: (currency: SfCurrency) => void;
+  setCustomer: (customer: SfCustomer) => void;
+  setLocale: (locale: string) => void;
+  setLocales: (locales: string[]) => void;
+}
+
+const DEFAULT_PROPS: SfStateProps = {
+  currencies: [],
+  currency: "",
+  locale: "",
+  locales: [],
+};
+
+const createSfState = (initialData?: Partial<SfStateProps>) => {
+  return createStore<SfState>()((set) => ({
+    cart: null,
+    customer: null,
+    ...DEFAULT_PROPS,
+    ...initialData,
+    setCart: (cart) => set({ cart }),
+    setCurrencies: (currencies) => set({ currencies }),
+    setCurrency: (currency) => set({ currency }),
+    setCustomer: (customer) => set({ customer }),
+    setLocale: (locale) => set({ locale }),
+    setLocales: (locales) => set({ locales }),
+  }));
+};
+
+export const SfStateContext = createContext<StoreApi<SfState> | null>(null);
+
+export function SfStateProvider({
+  children,
+  initialData,
+}: {
+  initialData: Partial<SfStateProps>;
+} & PropsWithChildren) {
+  const stateReference = useRef<StoreApi<SfState>>();
+
+  if (!stateReference.current) {
+    stateReference.current = createSfState(initialData);
+  }
+
+  return (
+    <SfStateContext.Provider value={stateReference.current}>
+      {children}
+    </SfStateContext.Provider>
+  );
+}
+
+function getSfContext() {
```

Comment:
```suggestion
function getSfStateContext() {
```

To be more consistent ;)